### DISCUSSION TITLE
[V5] Cannot call this.related inside a model

### DISCUSSION THREAD

**Discussion by yeboster** [adonisjs/core#1555] [Help] [answered]
Hello all,

First of all thanks to this awesome project, I really like the static type checking!
I'm trying to get the relationships from inside a model but it doesn't work because `ExtractModelRelation` doesn't work well when calling  on `this.`

<img width="732" alt="2020-09-04 alle 09 33 02" src="https://user-images.githubusercontent.com/23556525/92212349-2022aa00-ee92-11ea-8950-3d3b17aead84.png">

I've understood that the property `roles` is just an opaque type, so I cannot use it to get related roles, but then how to return the roles inside a User model ? Right now I'm passing the user instance to the static method, so it works well, but it's a little cumbersome.

Thank you,
Have a nice day.

**Comment by thetutlage** [comment]
A known issue with Typescript directly. I have already reported it https://github.com/microsoft/TypeScript/issues/37778

**Comment by yeboster** [comment, marked as answer]
Wow, I didn't expect that. I've seen you pinged the issue, thanks 👍 

For anyone who had the same problem, I've used this workaround to trick Typescript:
```typescript
public static async isAdmin(user: User) {
  const roles = await user.related('roles').query().select('name')
  return roles.some((role) => role.name == 'admin')
}

public async isAdmin() {
  return await User.isAdmin(this)
}
```

**Reply by keef3ar** [reply]
I have the same issue.
it works this way
```typescript
public getProjectById(project_id) {
    const user: user = this;
    return user
      .related("projects")
      .query()
      .where({ id: project_id })
      .first();
  }
